src/box/lua/console.lua

```diff
@@ -635,7 +635,7 @@ local function local_read(self)
             break
         end
         if delim == "" then
-            local lang = box.session.language
+            local lang = self.language
```

Comment:
> console: fixed lang identification in local_read()

Please, use the imperative mood in a commit message title. See point 6 at https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message

> NO_CHANGELOG=bugfix

We include bugfixes into release notes (see any release at https://github.com/tarantool/tarantool/releases).

> NO_TEST=bugfix

Do you suppose we shouldn't add regression tests?

AFAIU, the problem is invisible, because a complete SQL statement can't be interpreted as an incomplete Lua statement (at least I can't propose such example). So the problem is just that we run the lua related check for an SQL statement, extra work.

This invisibility is a reason to don't add a test and don't add a changelog entry. But 'bugfix' is not a reason.

Correct language is needed for further code change AFAIR. It is the motivation to fix it, right? Let's write the real motivation into the commit message.

Reply:
Fixed commit description.

Reply:
That's sad that the only way to find information why the change is needed and why it is invisible for a user is to look over review comments.

https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message

> 8. Use the body to explain **what and why** vs. how.

Reply:
I'm sorry, I am a bit lost. Do you think there should be reasoning about internal changes in continuation commit in here? For most people info that it is required is more than enough. The current description quite clearly states **what** bug was fixed, **why** we fix bugs is obvious.

Reply:
If it is a bug, there should be a reproducer of an incorrect behavior. But you describe it as 'invisible to user'. So it is not a bugfix, because there was no bug.

I still state that there is no clear description of the change.

Reply:
Agree with Alexander here: from my point of view, the patch looks definitely as a bugfix with the missing test. But it looks like we can't reproduce the issue, so it's not a bugfix, but kinda refactoring and so it would be nice to clarify the reasons of this change.

Reply:
Fixed:
```
    console: fix lang identification in local_read()
    
    Language in `local_read()` used to be set to `box.session.language` while
    the latter is always `nil` and `set_language()` sets `self.language`.
    
    Now the language in `local_read()` is identified correctly. This is
    required for performing continuation check (gh-4317) only in Lua mode.
    
    Needed for #4317
    
    NO_DOC=refactoring
    NO_CHANGELOG=refactoring
    NO_TEST=invisible to user
```